Added cron for delete old files

vitiodev opened the pull request.

vitiodev committed:
Added cron for delete old files

vitiodev committed:
Update readme.md

KSergiy commented on Cron/DeleteBackups.php:
```diff
@@ -0,0 +1,206 @@
+<?php
+declare(strict_types=1);
```
Between <?php and declare instruction must be one empty line

KSergiy commented on Cron/DeleteBackups.php:
```diff
@@ -0,0 +1,206 @@
+        DateTime $dateTime,
+        CronConfig $cronConfig,
+        LoggerInterface $logger
+    )
```
According to standards must be:
 ) {
in one line

KSergiy commented on Cron/DeleteBackups.php:
```diff
@@ -0,0 +1,206 @@
+     */
+    public function execute()
+    {
+        if ($this->cronConfig->isCronEnabled()) {
```
Better to do in a next way:
`
if (rule) {

 return null;

}

other code
`

KSergiy commented on Cron/DeleteBackups.php:
```diff
@@ -0,0 +1,206 @@
+            $cmsPath = $varPath . DS . CronConfig::CMS_DIR;
+            $directoryIterator = new \RecursiveDirectoryIterator($cmsPath, \FilesystemIterator::SKIP_DOTS);
+
+            if ($this->cronConfig->getMethodType() === CronConfig::PERIOD) {
```
Better to use Switch there and 2 separate functions for each case

KSergiy commented on Cron/DeleteBackups.php:
```diff
@@ -0,0 +1,206 @@
+            }
+        }
+
+        if (count($filesByWeeks)) {
```
Those ifs are excessive, better to move them into first ifs

KSergiy commented on Cron/DeleteBackups.php:
```diff
@@ -0,0 +1,206 @@
+     * @param $now
+     * @return array
+     */
+    private function findAndSortFilesByPeriods($directoryIterator, $now): array
```
In case used function return type then params in function must be typed too:
findAndSortFilesByPeriods(RecursiveDirectoryIterator $directoryIterator, string $now): array

KSergiy commented on Cron/DeleteBackups.php:
```diff
@@ -0,0 +1,206 @@
+    /**
+     * @param $directoryIterator
+     * @param $now
+     * @return array
```
Between `@return` and `@params` must be one empty line

KSergiy commented on Cron/DeleteBackups.php:
```diff
@@ -0,0 +1,206 @@
+        foreach ($files as $file) {
+            if (!$previousFile) $previousFile = $file;
+
+            if ($previousFile !== $file) {
```
Seems with this rule the first $file in the array will not be processed, is this expected?

KSergiy commented on Model/Config/Cron/Source/Periods.php:
```diff
@@ -0,0 +1,30 @@
+
+use Overdose\CMSContent\Model\Config\Cron\CronConfig;
+
+class Periods implements \Magento\Framework\Data\OptionSourceInterface
```
Better to use only class name instead of a full path

KSergiy commented on Model/Config/Cron/CronConfig.php:
```diff
@@ -0,0 +1,101 @@
+    /**
+     * @return mixed
+     */
+    public function getMethodType()
```
What type is expected in this function and in next, using mixed - bad practice

KSergiy commented on Cron/DeleteBackups.php:
```diff
@@ -0,0 +1,206 @@
+
+        foreach (new \RecursiveIteratorIterator($directoryIterator) as $file) {
+            $filePath = $file->getPathname();
+            if (strpos($filePath, CronConfig::HISTORY_DIR)) {
```
Better to make two separate ifs:
` 
 if (strpos($filePath, CronConfig::HISTORY_DIR)) { continue }

if (rule)
 `

KSergiy commented on Cron/DeleteBackups.php:
```diff
@@ -0,0 +1,206 @@
+            $now = $this->dateTime->gmtTimestamp();
+            $varPath = $this->directoryList->getPath(DirectoryList::VAR_DIR);
+            $cmsPath = $varPath . DS . CronConfig::CMS_DIR;
+            $directoryIterator = new \RecursiveDirectoryIterator($cmsPath, \FilesystemIterator::SKIP_DOTS);
```
Why not been used Magento framework functionality: Driver\File:readDirectory

SerhiiKoblia committed:
[d-b-f] Updated:

- refactored code

KSergiy committed:
[842741420] Updated:

- getting history directory duging creating history backups
- clearing backups

Task: https://www.wrike.com/open.htm?id=842741420
